### PR TITLE
share: share modal experiment

### PR DESCRIPTION
**Pull request by Zolmeister**

### CODE REVIEW

File: src/components/share_any_modal/index.coffee [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  ga? 'send', 'event', 'share_modal', 'share', 'kik'
+  kik?.send
+    title: "#{game.name}"
+    text: text
+    data:
+      gameKey: "#{game.key}"
+      share:
+        originUserId: me.id
+
+twitterShare = ({text, game, me}) ->
```
Comment (austinhallock) [inline src/components/share_any_modal/index.coffee]: this is the correct approach for the experiment. but if exp wins out, we may want to move the logic for these into a service so we don't have to repeat the code for everything

File: src/components/share_any_modal/index.coffee [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    href: href
+    redirect_uri: "http://#{config.HOST}"
+  }
+  shareUrl = "https://www.facebook.com/dialog/share?#{queryString}"
+
+  ga? 'send', 'event', 'share_modal', 'share', 'facebook'
+  UrlService.openWindow shareUrl
+
+module.exports = class ShareAnyModal
+  constructor: ({text, gameId}) ->
```
Comment (austinhallock) [inline src/components/share_any_modal/index.coffee]: text should be a prop

Comment (austinhallock) [inline src/components/share_any_modal/index.coffee]: also, pass the game object (not id) in (as a prop)

File: src/components/share_any_modal/index.styl [hunk 3]
```diff
@@ -0,0 +1,60 @@
+@require 'nib'
+
+json('../../stylus/vars.json')
+
+.z-share-any-modal
+
+  > .content
+    text-align: center
+    font-size: $fontSizeMedium
```
Comment (austinhallock) [inline src/components/share_any_modal/index.styl]: paper

File: src/components/share_any_modal/index.styl [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    background: $white
+
+    > .message
+      padding: 20px
+      font-size: $fontSizeMedium
```
Comment (austinhallock) [inline src/components/share_any_modal/index.styl]: ^

File: src/components/share_any_modal/index.styl [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+.z-share-any-modal
+
+  > .content
+    text-align: center
+    font-size: $fontSizeMedium
+    background: $white
+
+    > .message
+      padding: 20px
```
Comment (austinhallock) [inline src/components/share_any_modal/index.styl]: 16px (standard)

File: src/components/share_any_modal/index.styl [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+        > .kik
+          background-color: #83ba23
+
+    > .align-right
+      text-align: right
+
+      > .install-button
+        width: 150px
+        margin: 0 20px 20px 0
```
Comment (austinhallock) [inline src/components/share_any_modal/index.styl]: probably want 16px here, but i'd have to see

File: src/components/share_any_modal/index.styl [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        background: $green500
+        color: $white
```
Comment (austinhallock) [inline src/components/share_any_modal/index.styl]: $green500Text

File: src/components/share_any_modal/index.styl [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        > .twitter
+          background-color: #00aced
+
+        > .kik
+          background-color: #83ba23
+
+    > .align-right
+      text-align: right
+
+      > .install-button
```
Comment (austinhallock) [inline src/components/share_any_modal/index.styl]: rm

File: src/components/share_any_modal/index.coffee [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+  queryString = UrlLib.serializeQueryString {
+    app_id: config.FB_APP_ID
+    display: 'popup'
+    href: href
+    redirect_uri: "http://#{config.HOST}"
+  }
+  shareUrl = "https://www.facebook.com/dialog/share?#{queryString}"
+
+  ga? 'send', 'event', 'share_modal', 'share', 'facebook'
```
Comment (austinhallock) [inline src/components/share_any_modal/index.coffee]: we also need `ga? 'send', 'event', 'game', 'share', game.key` here (it's something we track in tao per game)

Comment (austinhallock) [inline src/components/share_any_modal/index.coffee]: (for twitter + kik too)

**Comment by Zolmeister** [comment]
updated

**Pull request merged**